modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java

```diff
@@ -1334,6 +1341,177 @@ public void defaultConverterCanHandleIncorrectType_2() {
+        StageLoader sl = new StageLoader(cb);
+
+        ListView listView = (ListView) ((ComboBoxListViewSkin)cb.getSkin()).getPopupContent();
+        ListViewBehavior lvBehavior = (ListViewBehavior)ControlSkinFactory.getBehavior(listView.getSkin());
+        InputMap<ListView<?>> lvInputMap = lvBehavior.getInputMap();
+        ObservableList<?> inputMappings = lvInputMap.getMappings();
+        // In ListViewBehavior KeyMappings for vertical orientation are added under 3rd child InputMap
+        InputMap<ListView<?>> verticalInputMap = lvInputMap.getChildInputMaps().get(2);
+        ObservableList<?> verticalInputMappings = verticalInputMap.getMappings();
+
+        cb.setEditable(true);
+        testKeyMappingsForEditableCB(inputMappings);
+        testCommonKeyMappings(inputMappings, verticalInputMappings);
+
+        cb.setEditable(false);
+        testKeyMappingsForNonEditableCB(inputMappings);
+        testCommonKeyMappings(inputMappings, verticalInputMappings);
+
+        sl.dispose();
+    }
+
+    private void testKeyMappingsForEditableCB(ObservableList<?> inputMappings) {
+        assertTrue(testInterceptor(inputMappings, new KeyBinding(KeyCode.HOME)));
+        assertTrue(testInterceptor(inputMappings, new KeyBinding(KeyCode.END)));
+    }
+
+    private void testKeyMappingsForNonEditableCB(ObservableList<?> inputMappings) {
+        assertFalse(testInterceptor(inputMappings, new KeyBinding(KeyCode.HOME)));
+        assertFalse(testInterceptor(inputMappings, new KeyBinding(KeyCode.END)));
+    }
+
+    private void testCommonKeyMappings(ObservableList<?> inputMappings,
+                                       ObservableList<?> verticalInputMappings) {
+        // Verify FocusTraversalInputMap
+        for(InputMap.Mapping<?> mapping : FocusTraversalInputMap.getFocusTraversalMappings()) {
+            assertFalse(inputMappings.contains(mapping));
+        }
+
+        // Verify default InputMap
+        assertTrue(testInterceptor(inputMappings, new KeyBinding(KeyCode.HOME).shift()));
+        assertTrue(testInterceptor(inputMappings, new KeyBinding(KeyCode.END).shift()));
+        assertTrue(testInterceptor(inputMappings, new KeyBinding(KeyCode.HOME).shortcut()));
+        assertTrue(testInterceptor(inputMappings, new KeyBinding(KeyCode.END).shortcut()));
+        assertTrue(testInterceptor(inputMappings, new KeyBinding(KeyCode.A).shortcut()));
+
+        // Verify vertical child InputMap
+        assertTrue(testInterceptor(verticalInputMappings, new KeyBinding(KeyCode.HOME).shortcut().shift()));
+        assertTrue(testInterceptor(verticalInputMappings, new KeyBinding(KeyCode.END).shortcut().shift()));
+    }
+
+    private boolean testInterceptor(ObservableList<?> mappings, KeyBinding binding) {
+        int i = mappings.indexOf(new KeyMapping(binding, null));
+        return ((KeyMapping)mappings.get(i)).getInterceptor().test(null);
+    }
+
```

Comment:
his throws an NPE without the fix - which makes the test using this method still fail (good!), but a bit unspecific for my taste. A slight re-organisation of the helpers might help: move the asserts down instead of returning a boolean.

Reply:
Change function as you suggested. Unlike ListViewTest, ComboBoxTest needs to verify if the interceptor correctly evaluates to either true or false, so it needed a minor tweak to use `assertEquals()`